File: src/main/java/cz/jskrabal/proxy/util/ContentType.java
package cz.jskrabal.proxy.util;

import java.util.Optional;

/**
 * Created by janskrabal on 03/06/16.
 */
public enum ContentType {
    IMAGE_JPEG("image/jpeg", "jpg"),
    IMAGE_GIF("image/gif", "gif"),
    IMAGE_PNG("image/png", "png");

    private String contentType;
    private String fileExtension;

    ContentType(String contentType, String fileExtension) {
        this.contentType = contentType;
        this.fileExtension = fileExtension;
    }

    public String getContentType() {
        return contentType;
    }

    public String getFileExtension() {
        return fileExtension;
    }

    public static Optional<ContentType> fromContentType(String contentType) {
        for(ContentType val: ContentType.values()){
            if(val.getContentType().equals(contentType)){
                return Optional.of(val);
            }
        }
        return Optional.empty();
    }
}

File: src/main/kotlin/cz/jskrabal/proxy/model/ContentType.kt
package cz.jskrabal.proxy.model

/**
 * Created by janskrabal on 03/06/16.
 */
enum class ContentType(val contentType: String, val fileExtension: String) {
    IMAGE_JPEG("image/jpeg", "jpg"),
    IMAGE_GIF("image/gif", "gif"),
    IMAGE_PNG("image/png", "png");


    companion object {
        fun fromContentType(contentType: String): ContentType? {
            return ContentType.values()
                    .firstOrNull { it.contentType == contentType }
        }

        fun fromExtension(extension: String): ContentType? {
            return ContentType.values()
                    .firstOrNull { it.fileExtension == extension }
        }
    }
}
